gnovm/stdlibs/std/emit_event.go

```diff
@@ -0,0 +1,36 @@
+package std
+
+// ref: https://github.com/gnolang/gno/pull/853
+
+import (
+	"fmt"
+	"time"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+)
+
+func X_emitEvent(m *gno.Machine, typ string, attrs []string) {
+	eventAttrs := make([]sdk.EventAttribute, len(attrs)/2)
+	pkgPath := CurrentRealmPath(m)
+
+	attrLen := len(attrs)
+	if attrLen%2 != 0 {
+		panic(fmt.Sprintf("attributes has an odd number of elements. current length: %d", attrLen))
+	}
+
+	for i := 0; i < attrLen; i += 2 {
+		eventAttrs[i/2] = sdk.EventAttribute{
+			Key:   attrs[i],
+			Value: attrs[i+1],
+		}
+	}
+
+	timestamp := time.Now().Unix()
```

Comment:
shouldn't this be the block timestamp?

Reply:
Yes. So, I modified it to get the `timestamp` data from the block header by adding `GetTimestamp` function. Thanks!